homeassistant/components/wilight/parent_device.py

```diff
@@ -0,0 +1,110 @@
+"""The WiLight Device integration."""
+import asyncio
+import logging
+
+import pywilight
+import requests
+
+from homeassistant.const import CONF_HOST, EVENT_HOMEASSISTANT_STOP
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class WiLightParent:
+    """Manages a single WiLight Parent Device."""
+
+    def __init__(self, hass, config_entry):
+        """Initialize the system."""
+        self._host = config_entry.data[CONF_HOST]
+        self._hass = hass
+        self._api = None
+
+    @property
+    def host(self):
+        """Return the host of this parent."""
+        return self._host
+
+    @property
+    def api(self):
+        """Return the api of this parent."""
+        return self._api
+
+    async def async_setup(self):
+        """Set up a WiLight Parent Device based on host parameter."""
+        host = self._host
+        hass = self._hass
+
+        api_device = await hass.async_add_executor_job(create_api_device, host, 45995)
+
+        if api_device is None:
+            return False
+
+        @callback
+        def disconnected():
+            # Schedule reconnect after connection has been lost.
+            _LOGGER.warning("WiLight %s disconnected", api_device.device_id)
+            async_dispatcher_send(
+                hass, f"wilight_device_available_{api_device.device_id}", False
+            )
+
+        @callback
+        def reconnected():
+            # Schedule reconnect after connection has been lost.
+            _LOGGER.warning("WiLight %s reconnect", api_device.device_id)
+            async_dispatcher_send(
+                hass, f"wilight_device_available_{api_device.device_id}", True
+            )
+
+        async def connect(api_device):
+            # Set up connection and hook it into HA for reconnect/shutdown.
+            _LOGGER.info("Initiating WiLight connection to %s", api_device.device_id)
+
+            client = await api_device.config_client(
+                disconnect_callback=disconnected,
+                reconnect_callback=reconnected,
+                loop=asyncio.get_running_loop(),
+                logger=_LOGGER,
+            )
+
+            # handle shutdown of WiLight asyncio transport
+            hass.bus.async_listen_once(
+                EVENT_HOMEASSISTANT_STOP, lambda x: client.stop()
+            )
+
+            _LOGGER.info("Connected to WiLight device: %s", api_device.device_id)
+
+        await connect(api_device)
+
+        self._api = api_device
+
+        return True
+
+    async def async_reset(self):
+        """Reset api."""
+
+        # If the initialization was wrong.
+        if self._api is None:
+            return True
+
+        self._api.client.stop()
+
+
+def create_api_device(host, port):
+    """Create an API Device."""
+    url = f"http://{host}:{port}/wilight.xml"
```

Comment:
Why is this string not created in the library? This is protocol level information.

Reply:
I could create a new function in the API like:
device = pywilight.device_from_host_port (host, port)

But I would have to do a new revision in the API too.

If this is not a major inconvenience, could it be in a new PullRequest when adding new components?

Reply:
This PR isn't merged yet. We can wait for the library bump.

Reply:
I will modify the library to support this.